src/main/java/com/google/sps/servlets/ChallengesServlet.java

```diff
@@ -0,0 +1,48 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import com.google.sps.data.ChallengeData;
+import java.io.IOException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Scanner;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that manages challenges **/
+@WebServlet("/challenges")
+public class ChallengesServlet extends HttpServlet {
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      String json = convertToJsonUsingGson();
+      response.setContentType("application/json;");
+      response.getWriter().println(json);
+    }
+
+   /* @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    } */
+
+    private String convertToJsonUsingGson(){
```

Comment:
There's no need to put this in a separate method. Just put the code here in doGet. However, if you plan to call convertToJsonUsingGson for converting other objects (e.g. in your future doPost method implementation), I'd suggest you make convertToJsonUsingGson take a generic object, rather than hard coding the conversion of ChallengeData.challenges.